src/WattleScript.Interpreter/Tree/Statements/ClassDefinitionStatement.cs

```diff
@@ -116,13 +145,24 @@ public ClassDefinitionStatement(ScriptLoadingContext lcontext) : base(lcontext)
                         switch (lcontext.Lexer.Current.Type)
                         {
                             case TokenType.Brk_Open_Round:
-                                functions.Add((T.Text, new FunctionDefinitionExpression(lcontext, SelfType.Implicit, false)));
+                                
+                                if (functions.ContainsKey(T.Text))
+                                {
+                                    throw new SyntaxErrorException(T, "duplicate declaration of a function '{0}'", T.Text);
+                                }
+                                
+                                functions.Add(T.Text, new MemberFieldInfo(T.Text, new FunctionDefinitionExpression(lcontext, SelfType.Implicit, false, modifierFlags), modifierFlags));
```

Comment:
change SelfType to SelfType.None if function is static

Reply:
resolved in ab2872c